banzai/main.py

```diff
@@ -86,145 +90,154 @@ def get_stages_todo(last_stage=None, extra_stages=None):
-    original_filename = pipeline_context.filename
-    for image in image_list:
-        pipeline_context.filename = os.path.basename(image)
-        try:
-            run(stages_to_do, pipeline_context, image_types=image_types)
-        except Exception as e:
-            logger.error('{0}'.format(e), extra_tags={'filename': pipeline_context.filename,
-                                                          'filepath': pipeline_context.raw_path})
-    pipeline_context.filename = original_filename
+        if raw_path is None:
+            raw_path = pipeline_context.raw_path
+        return pipeline_context, raw_path
 
 
-def reduce_science_frames_console():
-    run_end_of_night_from_console([reduce_science_frames], IMAGING_CRITERIA)
+def make_master_bias(pipeline_context=None, raw_path=None):
+    pipeline_context, raw_path = parse_directory_args(pipeline_context, raw_path)
+    process_directory(pipeline_context, raw_path, ['BIAS'], last_stage=trim.Trimmer,
+                      extra_stages=[bias.BiasMasterLevelSubtractor, bias.BiasComparer, bias.BiasMaker],
+                      log_message='Making Master BIAS', calibration_maker=True)
 
 
-def reduce_trailed_frames(pipeline_context):
-    stages_to_do = get_stages_todo()
-    reduce_frames_one_by_one(stages_to_do, pipeline_context, image_types=['TRAILED'])
+def make_master_dark(pipeline_context=None, raw_path=None):
+    pipeline_context, raw_path = parse_directory_args(pipeline_context, raw_path)
+    process_directory(pipeline_context, ['DARK'], last_stage=bias.BiasSubtractor,
+                      extra_stages=[dark.DarkNormalizer, dark.DarkComparer, dark.DarkMaker],
+                      log_message='Making Master Dark', calibration_maker=True, raw_path=raw_path)
 
 
-def reduce_trailed_frames_console():
-    run_end_of_night_from_console([reduce_trailed_frames], IMAGING_CRITERIA)
+def make_master_flat(pipeline_context=None, raw_path=None):
+    pipeline_context, raw_path = parse_directory_args(pipeline_context, raw_path)
+    process_directory(pipeline_context, ['SKYFLAT'], last_stage=dark.DarkSubtractor, log_message='Making Master Flat',
+                      extra_stages=[flats.FlatNormalizer, qc.PatternNoiseDetector, flats.FlatComparer, flats.FlatMaker],
+                      calibration_maker=True, raw_path=raw_path)
 
 
-def preprocess_sinistro_frames(pipeline_context):
-    stages_to_do = get_stages_todo(mosaic.MosaicCreator)
-    reduce_frames_one_by_one(stages_to_do, pipeline_context, image_types=['EXPOSE', 'STANDARD',
-                                                                          'BIAS', 'DARK', 'SKYFLAT',
-                                                                          'TRAILED', 'EXPERIMENTAL'])
+def reduce_science_frames(pipeline_context=None, raw_path=None):
+    pipeline_context, raw_path = parse_directory_args(pipeline_context, raw_path)
+    process_directory(IMAGING_CRITERIA, ['EXPOSE', 'STANDARD'], raw_path=raw_path)
```

Comment:
`pipeline_context` should be passed to `process_directory`, also for the methods following this one